### PR TITLE
Add CF CLI output to VSTS variable so it can be used in the pipeline

### PR DESCRIPTION
**Pull request by Geertvdc**
When you run a CLI command it can be often useful to get it's output back to a variable so you can use it later in your pipeline. I've added this by adding a output variable in the VSTS task.json and updated the typescript code to write back the CLI output back to this output variable.

I've tested it by creating my own task based on this one in my own pipeline.

### CODE REVIEW

File: Extension/CloudFoundryCLI/cloudfoundrycli.ts [hunk 1]
```diff
@@ -65,10 +65,16 @@ if (!cfPath) {
             if (args) {
                 cfCmd.line(args);
             }
-            cfCmd.exec()
-                .fail(function (err) {
-                    tl.setResult(tl.TaskResult.Failed, '' + err);
-                });
+            
+            var cfResult = cfCmd.execSync();
```
Comment (madhurig) [inline Extension/CloudFoundryCLI/cloudfoundrycli.ts:69]: Only short running commands should be run synchronously. cf CLI should be run async. Please switch back to using `cfCmd.exec()`

File: Extension/CloudFoundryCLI/cloudfoundrycli.ts [hunk 2]
```diff
@@ -65,10 +65,16 @@ if (!cfPath) {
...
+            
+            if(cfResult.error != null){
+                tl.setResult(tl.TaskResult.Failed, '' + cfResult.error);
+            }
+            else{
+                tl.setVariable("CfCliOutput", cfResult.stdout, false);
```
Comment (madhurig) [inline Extension/CloudFoundryCLI/cloudfoundrycli.ts:75]: Variables should be small actionable things we set for the next steps or pipeline to use. The entire stdout from a command is excessive. Generally variables are say a version that was deployed or the path to deployment file. The variables are stored in the system environment and the entire output might be fit in the limits different OSes have as well.

**Review by madhurig — CHANGES_REQUESTED** [review]
Thank you for your contribution. Please see my comments. We apologize for the delay in addressing this PR.

File: Extension/CloudFoundryCLI/cloudfoundrycli.ts (hunk 2 above)
Comment (Geertvdc) [reply Extension/CloudFoundryCLI/cloudfoundrycli.ts:75]: but you don't know what the command is so you can't get the right env variable. I added this so if use the CLI to query something you can actually use its output. do you have any other ideas how to do that then?

Comment (madhurig) [reply Extension/CloudFoundryCLI/cloudfoundrycli.ts:75]: I am not sure I follow the scenario, can you provide more detail? Wouldn't parsing the cfCommand input help with figuring out which command is being run. What data is of interest? Can you parse the stdout and avoid setting the variable if what you are looking for is not present?

**Comment by madhurig** [comment]
Closing since this an old PR with no updates and we don't want the entire CLI output to be set as a variable.

**Pull request closed without merging**